packages/craco-esbuild/src/index.js

```diff
@@ -48,11 +53,49 @@ module.exports = {
    * To process the js/ts files we replace the babel-loader with the esbuild jest loader
    */
   overrideJestConfig: ({ jestConfig }) => {
-    // TODO add back once https://github.com/aelbore/esbuild-jest/issues/7 is fixed
-    // // Replace babel transform with esbuild
-    // const key = Object.keys(jestConfig.transform)[0];
-    // jestConfig.transform[key] = [require.resolve("esbuild-jest")];
+    const options = {
+      loaders: {
+        '.js': 'jsx',
+        '.test.js': 'jsx',
+        '.ts': 'tsx',
+        '.test.ts': 'tsx'
+      },
+    };
+
+    // Replace babel transform with esbuild
+    // babelTransform is first transformer key
+    /* 
+    transform:
+      {
+        '^.+\\.(js|jsx|mjs|cjs|ts|tsx)$': 'node_modules\\react-scripts\\config\\jest\\babelTransform.js',
+        '^.+\\.css$': 'node_modules\\react-scripts\\config\\jest\\cssTransform.js',
+        '^(?!.*\\.(js|jsx|mjs|cjs|ts|tsx|css|json)$)': 'node_modules\\react-scripts\\config\\jest\\fileTransform.js'
+      }
+    */
+    const babelKey = Object.keys(jestConfig.transform)[0];
+
+    // We replace babelTransform and add loaders to esbuild-jest
+    jestConfig.transform[babelKey] = [require.resolve("esbuild-jest"), options];
+    
+    // Adds loader to all other transform options (2 in this case: cssTransform and fileTransform)
```

Comment:
🤯 looks like it's working, can you just add a link to the issue you created on esbuild-jest so I can keep track?

Reply:
Link: https://github.com/aelbore/esbuild-jest/issues/18